include header 'wayland-client.h' not found on Linux by default

When compiling with the wayland-client package on Linux (openSUSE, the corresponding headers are installed in a subdirectory of /usr/include. This path is not added to the compiler search path by openxr, so the required file 'wayland-client.h' is not found.

How to reproduce
--
When running:
``` 
$ sudo zypper install cmake gcc-c++ git "pkgconfig(wayland-client)" "pkgconfig(xcb)" "pkgconfig(x11)" "pkgconfig(xrandr)" "pkgconfig(xxf86vm)" "pkgconfig(jsoncpp)" "pkgconfig(vulkan)" vulkan-headers Mesa-libGL-devel
$git clone https://github.com/KhronosGroup/OpenXR-SDK-Source.git
$ cd OpenXR-SDK-Source
$ cmake -B build
$ cmake --build build
``` 
I get:
```
In file included from /home/user/src/vsgFramework-build/components/vsgvr-src/deps/openxr/src/common/platform_utils.hpp:12:0,
                 from /home/user/src/vsgFramework-build/components/vsgvr-src/deps/openxr/src/common/filesystem_utils.cpp:13:
/home/user/src/vsgFramework-build/components/vsgvr-src/deps/openxr/src/common/xr_dependencies.h:88:10: fatal error: wayland-client.h: file not found
 #include "wayland-client.h"
          ^~~~~~~~~~~~~~~~~~
```
Additional information
--
This looks to be a generic problem on linux as packages that generate binaries for openxr have added some workarounds to work around this problem, see [patching the header](https://build.opensuse.org/package/view_file/home:leviathanch:vr/openxr-sdk-source/includefix.patch?expand=1) and [adding additional include path](https://build.opensuse.org/package/view_file/hardware:xr/OpenXR-SDK/OpenXR-SDK.spec?expand=1)

--- a/changes/sdk/pr.346.gh.OpenXR-SDK-Source.md
+++ b/changes/sdk/pr.346.gh.OpenXR-SDK-Source.md
@@ -1,0 +1,1 @@
+loader,api_layers: Fix finding wayland-client.h on linux

--- a/src/api_layers/CMakeLists.txt
+++ b/src/api_layers/CMakeLists.txt
@@ -104,6 +104,10 @@ if(Vulkan_FOUND)
         PRIVATE ${Vulkan_INCLUDE_DIRS}
     )
 endif()
+if(BUILD_WITH_WAYLAND_HEADERS)
+    target_include_directories(XrApiLayer_api_dump
+        PRIVATE ${WAYLAND_CLIENT_INCLUDE_DIRS})
+endif()
 
 # Basics for core_validation API Layer
 
@@ -167,6 +171,10 @@ if(Vulkan_FOUND)
     target_include_directories(XrApiLayer_core_validation
         PRIVATE ${Vulkan_INCLUDE_DIRS}
     )
+endif()
+if(BUILD_WITH_WAYLAND_HEADERS)
+    target_include_directories(XrApiLayer_core_validation
+        PRIVATE ${WAYLAND_CLIENT_INCLUDE_DIRS})
 endif()
 
 if(WIN32)

--- a/src/cmake/presentation.cmake
+++ b/src/cmake/presentation.cmake
@@ -132,7 +132,7 @@ elseif(PRESENTATION_BACKEND MATCHES "wayland")
             ${WAYLAND_PROTOCOLS_DIR}/${PROTOCOL}.h
         )
 
-        target_include_directories(openxr-gfxwrapper PUBLIC ${WAYLAND_PROTOCOLS_DIR})
+        target_include_directories(openxr-gfxwrapper PUBLIC ${WAYLAND_PROTOCOLS_DIR} ${WAYLAND_CLIENT_INCLUDE_DIRS})
         target_link_libraries(
             openxr-gfxwrapper PRIVATE ${EGL_LIBRARIES} ${WAYLAND_CLIENT_LIBRARIES} ${WAYLAND_EGL_LIBRARIES}
         )

--- a/src/loader/CMakeLists.txt
+++ b/src/loader/CMakeLists.txt
@@ -90,6 +90,10 @@ else()
         target_compile_options(openxr_loader PRIVATE -Wno-unused-parameter)
     endif()
 endif()
+if(BUILD_WITH_WAYLAND_HEADERS)
+    target_include_directories(openxr_loader PRIVATE ${WAYLAND_CLIENT_INCLUDE_DIRS})
+endif()
+
 set_target_properties(openxr_loader PROPERTIES FOLDER ${LOADER_FOLDER})
 
 if(LOADER_EXTERNAL_GEN_DEPENDS)

